### PR TITLE
fix: usage of `node12` in `actions/checkout`

### LINKED ISSUE #90: CI `uses: actions/checkout@v2` which `uses node12 which is deprecated`

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #90.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -9,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
```
Comment (hamirmahal) [inline .github/workflows/test.yml:11]: This is the latest version at the time of this writing.

Comment (hamirmahal) [inline .github/workflows/test.yml:11]: This is the latest major version at the time of this writing.

**Review by y21 — APPROVED** [review]
Thanks!

**Comment by hamirmahal** [comment]
You're welcome!

**Pull request merged**